test/transforms/test_random_link_split.py

```diff
@@ -196,3 +196,16 @@ def test_random_link_split_on_undirected_hetero_data():
                                 rev_edge_types=('p', 'p'))
     train_data, val_data, test_data = transform(data)
     assert train_data['p', 'p'].is_undirected()
+
+
+def test_random_link_split_insufficient_negative_edges():
+    data = Data()
+    data.edge_index = torch.tensor([[0, 0, 0, 1, 1, 2, 2],
+                                    [1, 2, 3, 0, 2, 0, 1]])
+    transform = RandomLinkSplit(num_val=0.33, num_test=0.33,
+                                is_undirected=False, neg_sampling_ratio=2,
+                                split_labels=True)
+    train_data, val_data, test_data = transform(data)
+    assert len(train_data.neg_edge_label) != 0
+    assert len(val_data.neg_edge_label) != 0
+    assert len(test_data.neg_edge_label) != 0
```

Comment:
Can we check here for the actual returned number of edges? I guess it may be
```python
assert len(train_data.neg_edge_label) == 1
assert len(val_data.neg_edge_label) == 1
assert len(test_data.neg_edge_label) == 0
```

Reply:
They're all non-zero, but yes we can check the exact size, I'll add that.